src/main/java/io/nats/client/impl/NatsMessageConsumerBase.java

```diff
@@ -101,21 +101,19 @@ public void stop() {
 
     @Override
     public void close() throws Exception {
-        lenientClose();
+        stopped.set(true);
+        shutdownSub();
```

Comment:
I've clarified this behavior.
* set stopped to true since we are close (same as stop() does)
* shut the sub down.
I do not set finished to true because their could be messages already at the client by not seen by the user.